### DISCUSSION TITLE
Feature: Course Survey

### DISCUSSION THREAD

**Discussion by CBID2** [open-sauced/intro#110] [Ideas]
### Type of feature

🍕 Feature

### Current behavior

I noticed that this course has not received a lot of feedback publicly, which in my experience as an educator, can make it somewhat difficult to make the needed updates or revisions.  

### Suggested solution

I think we should consider adding a link to a survey that people can take at the end of this course. That way, we have specific data on how helpful the course has been in open source, demographics of people that take it, and ultimately, suggestions on how to improve it or what topics need more focus. 
Since you used to be a teacher @BekahHW, I'd love to hear your thoughts on this.   

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by BekahHW** [comment]
I like this idea a lot. 

I have a survey for folks who have taken the workshop and have received positive feedback. 

As an open source project, I'm more inclined to open up a discussion for feedback and link it in the contributing guide, readme, and at the end of the course. What do you think about that?

**Comment by CBID2** [comment]
> I like this idea a lot.
> 
> I have a survey for folks who have taken the workshop and have received positive feedback.
> 
> As an open source project, I'm more inclined to open up a discussion for feedback and link it in the contributing guide, readme, and at the end of the course. What do you think about that?
That's cool @BekahHW! :) Maybe you can post that survey too? While a discussion is helpful, I think a survey would be best because it is something we can refer back to when raising issues or submitting PRs.

**Comment by BekahHW** [comment]
Can you be more specific about the types of things you'd find valuable in raising issues and PRs?

**Comment by CBID2** [comment]
> Can you be more specific about the types of things you'd find valuable in raising issues and PRs?

To elaborate, I mean say 50% of people on the survey found the subsection on branching difficult to understand". From there, we can raise an issue and submit a pull request that addresses said problem. Does that make more sense now @BekahHW?

**Comment by BekahHW** [comment]
@CBID2, I think that could be handled in the discussions, promote transparency, and encourage others to provide feedback. Folks could also upvote anything they agree with to give us those numbers as well.

**Comment by CBID2** [comment]
> @CBID2, I think that could be handled in the discussions, promote transparency, and encourage others to provide feedback. Folks could also upvote anything they agree with to give us those numbers as well. 

That's true @BekahHW, & we can link it to the end of the course.

**Comment by BekahHW** [comment]
@CBID2 we can get started with [this](https://github.com/open-sauced/intro/discussions/categories/feedback).

**Comment by CBID2** [comment]
> @CBID2 we can get started with [this](https://github.com/open-sauced/intro/discussions/categories/feedback).

That's a good start @BekahHW, but I feel we need something there. Like questions already in place. Something...[like this](https://www.questionpro.com/survey-templates/student-course-evaluation-teacher-evaluation/) but less academic.

**Comment by BekahHW** [comment]
@CBID2 Can you investigate how other tech projects are doing it? In terms of audience expectation, we should see other than discussions and issues, what people are doing. My main concern with added extra things is we move away from the open source ecosystem and transparency. It's also one more thing to keep track of.

Astro's feedback form is pretty general too:

![What’s on your mind? form](https://github.com/open-sauced/intro/assets/34313413/a970575e-29e5-40fd-a4fb-4c1ba4ffa4b8)

**Comment by CBID2** [comment]
> @CBID2 Can you investigate how other tech projects are doing it? In terms of audience expectation, we should see other than discussions and issues, what people are doing. My main concern with added extra things is we move away from the open source ecosystem and transparency. It's also one more thing to keep track of.
> 
> 
> 
> Astro's feedback form is pretty general too:
> 
> 
> 
> ![What’s on your mind? form](https://github.com/open-sauced/intro/assets/34313413/a970575e-29e5-40fd-a4fb-4c1ba4ffa4b8)
> 
> 

Sure @BekahHW!

**Comment by CBID2** [comment]
Hey @BekahHW! :) Here's a survey that The Good Docs Project uses to track their audience and they always post the results in their Slack channel: https://docs.google.com/forms/d/e/1FAIpQLSfPXlFXtHNVlLuJ6hReLE1oW_nL14BdVe-UAr1Ry_uxreIZQw/viewform
Chaoss provides some suggestions too: https://chaoss.community/kb/metric-contributors/

**Comment by Saduniperera131629** [comment]
-- This is an example lua script.
-- Use the file manager or the top left menu to create new lua files.

print("Hello, World!")
print("world") 
print("Gallery"
```JavaScript
console.log("hello world");
console.log("There are 3 bugs on my house 🏠.");
``` print sending.................
print sended 
print("Gallery")
```JavaScript
console.log("hello world");
console.log("There are 3 bugs on my house 🏠.");
```print("There are 3 bug on my house 🏠")
